### PR TITLE
Add nightly build to CI; add new e2e tests

### PR DESCRIPTION
**Pull request by robertp**
Also update k8s version and switch to new Helm repo for e2e testing

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -114,7 +114,10 @@ setup-e2e-tests: anchore-ci venv ## Start kind cluster and set up end to end tes
 	@$(ACTIVATE_VENV) && $(CI_CMD) setup-e2e-tests "$(COMMIT_SHA)" "$(DEV_IMAGE_REPO)" "$(GIT_TAG)" "$(TEST_IMAGE_NAME)"
 
 test-e2e: anchore-ci venv ## Run end to end tests (assuming cluster is running and set up has been run)
-	@$(ACTIVATE_VENV) && $(CI_CMD) e2e-tests
+	@$(MAKE) test-cli
+
+test-cli: setup-test-infra venv ## Run cli-driven end to end tests (assuming cluster is running and set up has been run)
```
Comment (Btodhunter) [inline Makefile:119]: What the purpose of adding a target that invokes another target? Why not just keep everything in the `test-e2e` target?

File: .circleci/config.yml [hunk 2]
```diff
@@ -227,6 +227,61 @@ workflows:
...
 
+  nightly_build:
+    triggers:
+      - schedule:
+          cron: "0 10 * * *"
+          filters:
+            branches:
+              only:
+                - master
+    jobs:
```
Comment (Btodhunter) [inline .circleci/config.yml:238]: every job in this workflow should have `filters: *filter_nightly`

File: Makefile (hunk 1 above)
Comment (robertp) [reply Makefile:119]: Introducing the change gradually - although I think it doesn't matter, so I'll make the change as you suggest!

File: .circleci/config.yml (hunk 2 above)
Comment (robertp) [reply .circleci/config.yml:238]: Will do, thanks!

**Pull request merged**